### ISSUE TITLE
Unable to access the module for an employee without setting all permissions

### ISSUE THREAD

**Issue by dingedi** [PrestaShop/PrestaShop#30788]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

then in personalize I have
![image](https://user-images.githubusercontent.com/64697405/211502799-90c509e2-0641-4153-9899-9c8ee34c3fe7.png)

but the profile can't access (invalid token, compromised) it unless I add add and delete permissions, which I don't want.
it worked without problems on 1.7.8

### Steps to reproduce

1. go to permissions
2. choose a profile
3. set show/edit permissions for modules
4. attempt to access a module from an account with this profile

### PrestaShop version(s) where the bug happened

8.0 and 8.0.1

### PHP version(s) where the bug happened

8.1.13

**Comment by paulnoelcholot** [comment]
Hello @dingedi,

I have tested this issue and I don't see regression with 1.7.7.8

Can you give me a screen video to explain me the problem please?

Thanks

**Comment by dingedi** [comment]
PrestaShop 1.7.8.8 OK
https://user-images.githubusercontent.com/64697405/211590448-0495425b-29bd-4559-93e5-7d667e10a10e.mp4


PrestaShop 8 NOT OK
https://user-images.githubusercontent.com/64697405/211590551-4243eb63-aee2-4633-893e-ebb16bda04ad.mp4

**Comment by paulnoelcholot** [comment]
Hello,

I reproduce the issue with PrestaShop version 8.0.0, I'll add this to the backlog so it can be fixed.

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

https://watch.screencastify.com/v/ItHExgBT7e1u1zSlBKqn

Thank you

**Comment by jolelievre** [comment]
Hi @dingedi 

Just one question to be sure Is it really here that this should be configured? The permission page allows defining permissions for each module:
![Capture d’écran 2023-01-10 à 17 41 34](https://user-images.githubusercontent.com/13801017/211610835-7ad065f1-8398-47da-b038-f763a99c9514.png)

It seems to be more the place to handle this no? The permission shown in the description refer to the module management page (to install, enable, ... modules).

TLDR; are you configuring the access of a specific module for your profile or the access to module management in read only?

**Comment by dingedi** [comment]
yes I have set the permissions to see/modify for the module (you can see in the video), I have not changed anything because the store was under prestashop 1.7.8 and since the update to PrestaShop 8 it no longer works

by adding the add/remove permissions for Modules > Modules Manager > Modules manager this solves the problem but it allows the user to install modules so it is not the solution

**Comment by dingedi** [comment]
I think the problem comes from this line  which requires having permissions for all actions to access a module
https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php#L103

related to this: https://github.com/PrestaShop/PrestaShop/pull/29067

**Comment by hibatallahAouadni** [comment]
Hello @dingedi 

Could you please check this issue #9786 seems similar to yours.
Waiting for your feedback.

Thanks!

**Comment by dingedi** [comment]
hello @hibatallahAouadni this issue seems very old and talking about ps 1.6 and 1.7, mine talks about a regression in prestashop 8 and 8.0.1 compared to 1.7.8.8
